src/Map.cpp

```diff
@@ -67,20 +67,20 @@ void cMap::Tick()
 
 void cMap::UpdateRadius(int a_PixelX, int a_PixelZ, unsigned int a_Radius)
 {
-	int PixelRadius = a_Radius / GetPixelWidth();
+	int PixelRadius = static_cast<int>(a_Radius / GetPixelWidth());
 
-	unsigned int StartX = Clamp(a_PixelX - PixelRadius, 0, (int)m_Width);
-	unsigned int StartZ = Clamp(a_PixelZ - PixelRadius, 0, (int)m_Height);
+	unsigned int StartX = static_cast<UInt32>(Clamp(a_PixelX - PixelRadius, 0, static_cast<int>(m_Width)));
+	unsigned int StartZ = static_cast<UInt32>(Clamp(a_PixelZ - PixelRadius, 0, static_cast<int>(m_Height)));
 
-	unsigned int EndX   = Clamp(a_PixelX + PixelRadius, 0, (int)m_Width);
-	unsigned int EndZ   = Clamp(a_PixelZ + PixelRadius, 0, (int)m_Height);
+	unsigned int EndX   = static_cast<UInt32>(Clamp(a_PixelX + PixelRadius, 0, static_cast<int>(m_Width)));
+	unsigned int EndZ   = static_cast<UInt32>(Clamp(a_PixelZ + PixelRadius, 0, static_cast<int>(m_Height)));
 
 	for (unsigned int X = StartX; X < EndX; ++X)
 	{
 		for (unsigned int Z = StartZ; Z < EndZ; ++Z)
 		{
-			int dX = X - a_PixelX;
-			int dZ = Z - a_PixelZ;
+			int dX = static_cast<int>(X - static_cast<UInt32>(a_PixelX));
+			int dZ = static_cast<int>(Z - static_cast<UInt32>(a_PixelZ));
```

Comment:
This uses a lot of casts unnecessarily, can't we unify the types to a single one to avoid all the casts?